autoflip_calculators/text_detection_calculator.cc

```diff
@@ -0,0 +1,262 @@
+            float angle = anglesData[x];
+            float cosA = std::cos(angle);
+            float sinA = std::sin(angle);
+            float h = x0_data[x] + x2_data[x];
+            float w = x1_data[x] + x3_data[x];
+
+            cv::Point2f offset(offsetX + cosA * x1_data[x] + sinA * x2_data[x],
+                           offsetY - sinA * x1_data[x] + cosA * x2_data[x]);
+            cv::Point2f p1 = Point2f(-sinA * h, -cosA * h) + offset;
+            cv::Point2f p3 = Point2f(-cosA * w, sinA * w) + offset;
+
+            cv::RotatedRect r(0.5f * (p1 + p3), Size2f(w, h), -angle * 180.0f / (float)CV_PI);
+            detections.push_back(r);
+            confidences.push_back(score);
+        } // end for x
+    } // end for y
+}   
+
+::mediapipe::Status TextDetectionCalculator::Process(
+    mediapipe::CalculatorContext* cc) {
+  // What if !cc->Inputs.HasTag("VIDEO")
+  if (cc->Inputs().HasTag("VIDEO") &&
+      cc->Inputs().Tag("VIDEO").Value().IsEmpty()) {
+    return ::mediapipe::UnknownErrorBuilder(MEDIAPIPE_LOC)
+           << "No VIDEO input at time " << cc->InputTimestamp().Seconds();
+  }
+
+  if (cc->Inputs().HasTag("VIDEO")) {
+    cv::Mat frame, blob;
+    frame = mediapipe::formats::MatView(
+        &cc->Inputs().Tag("VIDEO").Get<ImageFrame>());
+    frame_width_ = frame.cols;
+    frame_height_ = frame.rows;
+
+    // Mean subtraction and scalling.
+    blobFromImage(frame, blob, 1.0, Size(frame_width_, frame_height_), 
+                  Scalar(123.68, 116.78, 103.94), true, false);
+    // Detect the text.
+    detector.setInput(blob);
+    detector.forward(outs, outNames);
+
+    cv::Mat scores = outs[0];
+    cv::Mat geometry = outs[1];
+
+    // Decode predicted bounding boxes and corresponding confident scores.
+    std::vector<RotatedRect> boxes;
+    std::vector<float> confidences;
+    DecodeBoundingBoxes(scores, geometry, confidence_threshold, boxes, confidences);
+
+    // Apply non-maximum suppression procedure.
+    std::vector<int> indices;
+    cv::dnn::NMSBoxes(boxes, confidences, confidence_threshold, nms_threshold, indices);
+
+    auto region_set = ::absl::make_unique<DetectionSet>();
+    for (size_t i = 0; i < indices.size(); ++i) {
+      cv::RotatedRect& box = boxes[indices[i]];
+      cv::Rect2f RectBox = box.boundingRect();
+      float text_score = confidences[indices[i]];
+      
+      // Normalize the bounding box.
+      RectBox.x = NormalizeX(RectBox.x);
+      RectBox.y = NormalizeY(RectBox.y);
+      RectBox.width = NormalizeX(RectBox.width);
+      RectBox.height = NormalizeY(RectBox.height);
+      float x = std::max(0.0f, RectBox.x);
+      float y = std::max(0.0f, RectBox.y);
+      float width =
+          std::min(NormalizeX(RectBox.width) - x + RectBox.x, 1 - x);
+      float height = 
+          std::min(RectBox.height - y + RectBox.y, 1 - y);
+
+      // Convert the text bounding box to a region.
+      if (options_.export_text_bounding_box()) {
```

Comment:
I'm not sure this should be optional.  I think we want to require that we output the results.